src/app_model/registries/_keybindings_reg.py

```diff
@@ -133,13 +162,31 @@ def register_keybinding_rule(
                 command_id=id,
                 weight=rule.weight,
                 when=rule.when,
+                source=rule.source,
             )
             self._keybindings.append(entry)
+
+            # inverse map registry
+            kb = keybinding.to_int()
+            if kb not in self._keymap:
+                entries: list[_RegisteredKeyBinding] = []
+                self._keymap[kb] = entries
+            else:
+                entries = self._keymap[kb]
```

Comment:
after turning `_keymap` into a defaultdict, this becomes:
```suggestion
            entries = self._keymap[keybinding.to_int()]
```